build_tools/benchmarks/common/benchmark_definition.py

```diff
@@ -443,3 +443,113 @@ def from_json_str(json_str: str):
         BenchmarkRun.from_json_object(b) for b in json_object["benchmarks"]
     ]
     return results
+
+
+@dataclass(frozen=True)
+class CompilationInfo(object):
+  model_name: str
+  model_tags: Sequence[str]
+  model_source: str
+  target_arch: str
+  bench_mode: Sequence[str]
+
+  def __str__(self):
+    if self.model_tags:
+      tags = ",".join(self.model_tags)
+      model_part = f"{self.model_name} [{tags}] ({self.model_source})"
+    else:
+      model_part = f"{self.model_name} ({self.model_source})"
+    bench_mode_str = ",".join(self.bench_mode)
+    return f"{model_part} {self.target_arch} {bench_mode_str}"
+
+  def to_json_object(self) -> Dict[str, Any]:
+    return {
+        "model_name": self.model_name,
+        "model_tags": self.model_tags,
+        "model_source": self.model_source,
+        "target_arch": self.target_arch,
+        "bench_mode": self.bench_mode,
+    }
+
+  @staticmethod
+  def from_json_object(json_object: Dict[str, Any]):
+    return CompilationInfo(model_name=json_object["model_name"],
+                           model_tags=json_object["model_tags"],
+                           model_source=json_object["model_source"],
+                           target_arch=json_object["target_arch"],
+                           bench_mode=json_object["bench_mode"])
```

Comment:
I think this can just use dictionary expansion

```suggestion
    return CompilationInfo(**json_object)
```

Although at that point, it seems like the caller could just do that just as easily as well.